src/pages/data-management/index.jsx

```diff
@@ -32,14 +33,19 @@ const DataManagementPage = ({ route }) => {
   const existingExperiments = activeProject?.experiments
     .map((experimentId) => experiments[experimentId]);
 
+  const experimentsAreLoaded = activeProject?.experiments.every((experimentId) => experiments.ids.includes(experimentId));
```

Comment:
This is going to be very inefficient, it's an O(n^2) lookup. Could you convert at least one of them into a Set (e.g. `experiments.ids`) and use `Set.has`?